src/inferenceparameter.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+//*****************************************************************************
+// Copyright 2022 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <string>
+
+#include "pocapi.hpp"
+
+namespace ovms {
+
+// TODO should we own our own copy of value?
+class InferenceParameter {
+    const std::string name;
+    DataType datatype;
+    const std::string data;
```

Comment:
If we keep the parameter inside std::string then we cannot store zeros.

Reply:
Use std::array?

Reply:
proto's we use in TFS/KFS hold data in string with no problems.

Reply:
it's more convenient to use std::string::assign() that's why at this point in time I think it makes sense.